### PR TITLE
[OYPD-153] Update for paragrpah gallery, adds template, image style, view mode f…

### PR DESCRIPTION
**Pull request by david-hernandez**
Adds template, updates the paragraph display mode, adds a new display mode for images, and an image style.

screen shot 2017-01-30 at 5 08 57 pm

- [x] Add a page for a type that uses paragraphs, like landing page.
- [x] Add a paragraph of type "gallery".
- [x] Add multiple images and save the node.
- [x] Make sure it looks correct.

### CODE REVIEW

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build267

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/267/

**Comment by Sanchiz** [comment]
retest this please

File: themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+#}
+{%
+set classes = [
+  'paragraph',
+  'paragraph-gallery',
+  'paragraph--type--' ~ paragraph.bundle|clean_class,
+  view_mode ? 'paragraph--view-mode--' ~ view_mode|clean_class,
+]
+%}
+{{ kint(content.field_prgf_images) }}
```
Comment (alexschedrov) [inline themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig]: Please remove debug code.

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build271

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/271/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build271/profiles/openy/build/reports/behat

**Review by alexschedrov — CHANGES_REQUESTED** [review]

**Comment by alexschedrov** [comment]
1. Image style or CSS stretches the image vertically. Just compare original YGS image and build.
image
image

File: themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+{% if content.field_prgf_images['#items']|length >= 1 %}
+<div{{ attributes.addClass(classes) }}>
+  <div id="carousel" class="carousel slide" data-ride="carousel">
+    <div class="carousel-inner" role="listbox">
+      {% for key, value in content.field_prgf_images['#items'] %}
+      <div class="item{% if key == 0 %} active{% endif %}">
+        {{ content.field_prgf_images[key] }}
+        {% if content.field_prgf_images[key]['#media'].field_media_caption.0.value is not empty %}
```
Comment (podarok) [inline themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig:54]: Use set instruction with ternary operators to get values from the fields with default fallback
Using direct data in if prohibited by best practices (without defaults debugging in a future is a bit tough)
Example
```twig
{% set color = content.field_blog_color ? content.field_blog_color : 'red' %}
```

File: themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+              <div class="item__caption">
+                {{ content.field_prgf_images[key]['#media'].field_media_caption.0.value }}
+              </div>
+            </div>
+          </div>
+        {% endif %}
+      </div>
+      {% endfor %}
+    </div>
+    {% if content.field_prgf_images['#items']|length > 1 %}
```
Comment (podarok) [inline themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig:66]: Use set instruction with ternary operators to get values from the fields with default fallback
Using direct data in if prohibited by best practices (without defaults debugging in a future is a bit tough)
Example
```twig
{% set color = content.field_blog_color ? content.field_blog_color : 'red' %}
```

File: themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      <span class="fa fa-chevron-left" aria-hidden="true"></span>
+      <span class="sr-only">{{ 'Previous'|t }}</span>
+    </a>
+    <a class="right carousel-control" href="#carousel" role="button" data-slide="next">
+      <span class="fa fa-chevron-right" aria-hidden="true"></span>
+      <span class="sr-only">{{ 'Next'|t }}</span>
+    </a>
+    {% endif %}
+  </div>
+  {% if content.field_prgf_headline['#field_name'] is not empty %}
```
Comment (podarok) [inline themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig]: Use set instruction with ternary operators (at the top of the template) to get values from the fields with default fallback
Using direct data in if prohibited by best practices (without defaults debugging in a future is a bit tough)
Example
```twig
{% set color = content.field_blog_color ? content.field_blog_color : 'red' %}
```

File: themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+{%
+set classes = [
+  'paragraph',
+  'paragraph-gallery',
+  'paragraph--type--' ~ paragraph.bundle|clean_class,
+  view_mode ? 'paragraph--view-mode--' ~ view_mode|clean_class,
+]
+%}
+{{ kint(content.field_prgf_images) }}
+{% if content.field_prgf_images['#items']|length >= 1 %}
```
Comment (podarok) [inline themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig]: Use set instruction with ternary operators to get values from the fields with default fallback
Using direct data in if prohibited by best practices (without defaults debugging in a future is a bit tough)
Example
```twig
{% set color = content.field_blog_color ? content.field_blog_color : 'red' %}
```

**Review by podarok — CHANGES_REQUESTED** [review]

File: themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig (hunk 4 above)
Comment (david-hernandez) [reply themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig]: I don't understand what you are suggesting with these. There is no use for a ternary operator here. Nothing is being set. Can you clarify?

Comment (david-hernandez) [reply themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig]: I'm removing this if. The headline field is required so there is no need for it anyway.

**Comment by david-hernandez** [comment]
Something happened to the icon display mode for images. It is not a display mode that was part of this paragraph but was reused. It used to return an icon sized image using an image style. Now it returns "original image" which could be bad.

**Comment by gundevel** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build301

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/301/

**Comment by gundevel** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build301/profiles/openy/build/reports/behat

**Comment by Sanchiz** [comment]
retest this please

File: themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig (hunk 2 above)
Comment (alexschedrov) [reply themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig:54]: Let's fix this in follow up.

File: themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig (hunk 3 above)
Comment (alexschedrov) [reply themes/openy_themes/openy_rose/templates/paragraph/paragraph--gallery--default.html.twig:66]: Let's fix this in follow up.

**Comment by bjwilkins** [comment]
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build307

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/307/

**Comment by bjwilkins** [comment]
Behat test results file URL: http://openy-dev.ffwua.com/build307/profiles/openy/build/reports/behat

**Review by alexschedrov — APPROVED** [review]

**Pull request merged**